lib/Conversion/ArcToLLVM/LowerArcToLLVM.cpp

```diff
@@ -165,6 +166,34 @@ struct StorageGetOpLowering : public OpConversionPattern<arc::StorageGetOp> {
   }
 };
 
+struct AssertOpLowering : public OpConversionPattern<arc::AssertOp> {
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(arc::AssertOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const final {
+    auto condition = adaptor.getProperty();
+    auto module = op->getParentOfType<ModuleOp>();
+    rewriter.replaceOpWithNewOp<scf::IfOp>(
+        op, condition, [&](auto &builder, auto loc) {
```

Comment:
You probably intended to invert `condition` before passing it to the `scf.if` because the message should be printed if the assertion does not hold.